[rum] ConfigCat feature flag integration (#23741)
DataDog/documentation: content/en/real_user_monitoring/guide/setup-feature-flag-data-collection.md

Amplitude does not support this integration. Create a ticket with Amplitude to request this feature.

{{% /tab %}}
{{< /tabs >}}

### ConfigCat integration

{{< tabs >}}
{{% tab "Browser" %}}

> Initialize the ConfigCat Javascript SDK by subscribing to the `flagEvaluated` event that reports feature flag evaluations to Datadog:

Reviewer: **💬 suggestion:** ‏The phrasing looks weird to me. When reading this, I would understand that the "flagEvaluated" event has logic to report feature flag evaluations to Datadog out of the box. I don't think it describes the following code snippet correctly.

```suggestion
When initializing the ConfigCat Javascript SDK, subscribe to the `flagEvaluated` event and report feature flag evaluations to Datadog:
```

```javascript

>   const configCatClient = configcat.getClient(
>   '#YOUR-SDK-KEY#',
>   configcat.PollingMode.AutoPoll,
>   {
>     setupHooks: (hooks) =>
>       hooks.on('flagEvaluated', (details) => {
>         datadogRum.addFeatureFlagEvaluation(details.key, details.value);
>       })
>   }
>   );

Reviewer: **💬 suggestion:** ‏Indentation could be better

```suggestion
const configCatClient = configcat.getClient(
  '#YOUR-SDK-KEY#',
  configcat.PollingMode.AutoPoll,
  {
    setupHooks: (hooks) =>
      hooks.on('flagEvaluated', (details) => {
        datadogRum.addFeatureFlagEvaluation(details.key, details.value);
      })
  }
);
```

```

For more information about initializing the ConfigCat Javascript SDK, see ConfigCat's [JavaScript SDK documentation][1].

[1]: https://configcat.com/docs/sdk-reference/js

{{% /tab %}}
{{% tab "iOS" %}}

> Initialize the ConfigCat Swift iOS SDK by subscribing to the `flagEvaluated` event that reports feature flag evaluations to Datadog:

Author: ```suggestion
When initializing the ConfigCat iOS SDK, subscribe to the `flagEvaluated` event and report feature flag evaluations to Datadog:
```

```swift
  let client = ConfigCatClient.get(sdkKey: "#YOUR-SDK-KEY#") { options in
    options.hooks.addOnFlagEvaluated { details in
        RUMMonitor.shared().addFeatureFlagEvaluation(featureFlag: details.key, variation: details.value)
    }
  }
```

For more information about initializing the ConfigCat Swift (iOS) SDK, see ConfigCat's[Swift iOS SDK documentation][1].

[1]: https://configcat.com/docs/sdk-reference/ios

{{% /tab %}}
{{% tab "Android" %}}

> Initialize the ConfigCat Android SDK by subscribing to the `flagEvaluated` event that reports feature flag evaluations to Datadog:

Author: ```suggestion
When initializing the ConfigCat Android SDK, subscribe to the `flagEvaluated` event and report feature flag evaluations to Datadog:
```

```java
  ConfigCatClient client = ConfigCatClient.get("#YOUR-SDK-KEY#", options -> {
    options.hooks().addOnFlagEvaluated(details -> {
        GlobalRumMonitor.get().addFeatureFlagEvaluation(details.key, details.value);
    });
  });
```

For more information about initializing the ConfigCat Android SDK, see ConfigCat's [Android SDK documentation][1].

[1]: https://configcat.com/docs/sdk-reference/android

{{% /tab %}}
{{% tab "Flutter" %}}

> Initialize the ConfigCat Dart SDK by subscribing to the `flagEvaluated` event that reports feature flag evaluations to Datadog:

Author: ```suggestion
When initializing the ConfigCat Dart SDK, subscribe to the `flagEvaluated` event and report feature flag evaluations to Datadog:
```

```dart
  final client = ConfigCatClient.get(
    sdkKey: '#YOUR-SDK-KEY#',
    options: ConfigCatOptions(
        pollingMode: PollingMode.autoPoll(),
        hooks: Hooks(
            onFlagEvaluated: (details) => {
              DatadogSdk.instance.rum?.addFeatureFlagEvaluation(details.key, details.value);
            }
        )
    )
  );
```

For more information about initializing the ConfigCat Dart (Flutter) SDK, see ConfigCat's [Dart SDK documentation][1].

[1]: https://configcat.com/docs/sdk-reference/dart

{{% /tab %}}

{{% tab "React Native" %}}

> Initialize the ConfigCat React SDK by subscribing to the `flagEvaluated` event that reports feature flag evaluations to Datadog:

Author: ```suggestion
When initializing the ConfigCat React SDK, subscribe to the `flagEvaluated` event and report feature flag evaluations to Datadog:
```
